### PR TITLE
Add Proxy flavors 

### PR DESCRIPTION
**Pull request by leomassazza**

### CODE REVIEW

File: packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol [hunk 1]
```diff
@@ -0,0 +1,68 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./UniversalProxyImplementation.sol";
+import "../ownership/Ownable.sol";
+
+contract OwnableUUPSImplementation is UniversalProxyImplementation, Ownable {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol:7]: Rename `UniversalProxyImplementation` to `UUPSImplementation`?

File: packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    struct UnstructuredProxyStorage {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol:8]: You only need to destructure  a single variable in storage, the implementation address. That's it, no storage namespaces at all.

File: packages/core-contracts/contracts/proxy/UnstructuredProxy.sol [hunk 3]
```diff
@@ -0,0 +1,33 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ForwardingProxy.sol";
+
+contract UnstructuredProxy is ForwardingProxy {
+    struct UnstructuredProxyStorage {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/proxy/UnstructuredProxy.sol:7]: No need to use a namespace.

File: packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        address implementation;
+        bool simulatingUpgrade;
+    }
+
+    struct OwnableStorage {
```
Comment (ajsantander) [inline packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol:13]: Ownable should not be implemented at this level. For example, SNXToken.sol inherits UUPSImplementation and Ownable.

Comment (leomassazza) [reply packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol:13]: Yeah, that was my question on the call. if we needed it here or not

File: packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol (hunk 1 above)
Comment (leomassazza) [reply packages/core-contracts/contracts/proxy/OwnableUUPSImplementation.sol:7]: The UniversalProxyImplementation is storage agnostic. This Ownable (that doesn't need to be owneable according to the other thread) is the one that adds the unstructured storage

File: packages/core-contracts/contracts/proxy/UnstructuredProxy.sol (hunk 3 above)
Comment (i-stam) [reply packages/core-contracts/contracts/proxy/UnstructuredProxy.sol:7]: how are you going to return an address variable  in storage without using a namespace?

**Pull request closed without merging**